### PR TITLE
15571: do not fireOnRequestUpdate for fetch nodes when the pendingcs is not a fetch nodes

### PR DESCRIPTION
**Pull request by polmr**
This fixes issues with progress being reset before finalizing fetchnodes

### CODE REVIEW

File: src/megaapi_impl.cpp [hunk 1]
```diff
@@ -13834,6 +13834,10 @@ void MegaApiImpl::request_error(error e)
 
 void MegaApiImpl::request_response_progress(m_off_t currentProgress, m_off_t totalProgress)
 {
+    if (!client->isFetchingNodesPendingCS())
```
Comment (sergiohs84) [inline src/megaapi_impl.cpp:13837]: Maybe we can simply use here `MegaClient::fetchingnodes` directly??

File: src/request.cpp [hunk 2]
```diff
@@ -26,9 +26,19 @@
 
 namespace mega {
 
+bool Request::includesFetchingNodes() const
+{
+    return mIncludesFetchingNodes;
```
Comment (sergiohs84) [inline src/request.cpp:31]: I think we could avoid this flag by using a dynamic cast. Something like:
```suggestion
    return cmds.size() == 1 && dynamic_cast<CommandFetchNodes>(cmds.back());
```

File: src/megaapi_impl.cpp (hunk 1 above)
Comment (polmr) [reply src/megaapi_impl.cpp:13837]: > since the flag might be true while the response to other commands is processed
unfortunately, it is true indeed

Comment (polmr) [reply src/megaapi_impl.cpp:13837]: > MegaClient::exec() has access to the list of commands whose HTTP request's response is being processed.
I might be wrong, but I understand that for that we'd need to access RequestsDispatcher::inflightreq. That's why I saved the flag in pendingcs. Controling the flag before or inside `request_response_progess` won't make much difference I believe. (besides megacli no longer receiving it for other comands)

File: src/request.cpp (hunk 2 above)
Comment (polmr) [reply src/request.cpp:31]: nice one. updaing accordingly

**Pull request merged**